subsys/fs/fs.c

```diff
@@ -499,7 +499,9 @@ int fs_mkdir(const char *abs_path)
 	}
 
 	rc = mp->fs->mkdir(mp, abs_path);
-	if (rc < 0) {
+	if (rc == -EEXIST) {
+		LOG_WRN("directory already exists (%d)", rc);
+	} else if (rc < 0) {
```

Comment:
It is not a warning, it is an error. The operation actually failed.

Reply:
changed